### PR TITLE
Replace regex with password strength algorithm and add password strength bar

### PR DESCRIPTION
**Pull request by tmrd993**
## Summary of change
Removed the regex for the user password field and created a custom constraint to validate the password. Validation uses the password strength checking algorithm [zxcvbn4j](https://github.com/nulab/zxcvbn4j) to calculate a score between 0 and 5. Created an enum class that represents the password strength levels. Added a progress bar element to the Registration Form that represents a password strength bar. The various colors are defined in the custom-bar-styles.css file and can be changed at will.

updatedUI

## Related issue

https://github.com/knjk04/book-project/issues/262

## Pull request checklist

Please ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for a in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

### CODE REVIEW

**Review by ChFlick — APPROVED** [review]
Good work :+1: 

I'd suggest renaming PasswordDescriptor to PasswordStrength

File: src/main/java/com/karankumar/bookproject/backend/constraints/PasswordStrengthValidator.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    private PasswordStrength passwordStrength;
+    
+    @Override
+    public void initialize(PasswordStrengthCheck constraintAnnotation) {
+	this.passwordStrength = constraintAnnotation.value();
+    }
+
+    @Override
+    public boolean isValid(String password, ConstraintValidatorContext context) {
+	return new Zxcvbn().measure(password).getScore() >= passwordStrength.getStrengthNum();
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/backend/constraints/PasswordStrengthValidator.java:36]: Could you ensure that the bodies of each method are indented by 4 spaces?

**Review by knjk04 — CHANGES_REQUESTED** [review]
Thanks a lot for this! There's just a couple of small things to address

**Comment by knjk04** [comment]
And thanks for the review @ChFlick!

**Comment by tmrd993** [comment]
Done. I commited before merging your changes so there are two seperate commits now ( add copyright and requested changes). Sorry about that.

**Comment by knjk04** [comment]
No problem, thanks a lot!

**Pull request merged**